@amimodel/compileC.m

```diff
@@ -224,24 +224,26 @@ function compileC(this)
         end
     end
     
+    amiciSourceBaseNames = {'amici'; 'amici_interface_matlab'; 'amici_misc'};
+
     if(this.nxtrue == this.nx)
+        amiciSourceList = '';
+        for ii = 1:numel(amiciSourceBaseNames)
+            amiciSourceList = [amiciSourceList, '"', fullfile(this.wrap_path, 'src', [amiciSourceBaseNames{ii}, '.cpp']), '" '];
+        end
         eval(['mex ' DEBUG COPT ...
             ' -c -outdir ' fullfile(this.wrap_path,'src') ' ' ...
-            fullfile(this.wrap_path,'src','amici.cpp') ' ' ...
-            includesstr]);
+            amiciSourceList ' ' includesstr]);
     end
-        
-    if(this.nxtrue == this.nx)
```

Comment:
Yes, it was checked twice before. Before each compilation mex call. Now compilation is done in a single call.